src/OrbitCore/SymbolHelper.cpp

```diff
@@ -199,3 +199,44 @@ fs::path SymbolHelper::GenerateCachedFileName(const fs::path& file_path) const {
   auto file_name = absl::StrReplaceAll(file_path.string(), {{"/", "_"}});
   return cache_directory_ / file_name;
 }
+
+ErrorMessageOr<fs::path> SymbolHelper::FindDebugInfoFileLocally(std::string_view filename,
+                                                                uint32_t checksum) const {
+  std::set<fs::path> search_paths;
+  for (const auto& directory : symbols_file_directories_) {
+    search_paths.insert(directory / filename);
+  }
+
+  LOG("Trying to find debuginfo file with filename \"%s\"", filename);
+  for (const auto& debuginfo_file_path : search_paths) {
+    std::error_code error;
+    bool exists = fs::exists(debuginfo_file_path, error);
+    if (error) {
+      ERROR("Unable to stat \"%s\": %s", debuginfo_file_path.string(), error.message());
+      continue;
```

Comment:
To consider, but totally fine as-is: I always find "continue" a bit more problematic than a return in a function. Maybe move the part in the loop into a function that returns true/false if the file was successfully found? Then all these continues become 'return false'.

Reply:
clang-tidy agrees with you. Too complex. Too many nestings. I will change it. Thanks!

Reply:
done.